_drafts/schematic_review_checklist_for_firmware_engineer.md

```diff
@@ -0,0 +1,132 @@
+
+Some pre-schematic review checklist items:
+
+ - [ ] Get invited to the schematic review. Really, this is a callout to be working with your electrical engineer. If you've been working with your electrical engineer, providing guidance on what hardware you'll need, being invited to the schematic review should be a given.
+ - [ ] Build your checklist up front. Use this blog post as a starting point, but make sure you customize the list for the given project.
+
+## My current checklist
+Here's my current checklist. I didn't list all the stories as above (this would've made way to long of a checklist), but I hope it's helpful for others!
+
+### Power/Battery
+
+- [ ] How does the firmware power off the device? Especially for a battery powered device.[1](#story-1-brown-out-loop)
+- [ ] For a battery powered device, Can the firmware measure how much power is available? How much accuracy is needed in this measurement? [1](#story-1-brown-out-loop)
+- [ ] How does an user reset your device?[2](#story-2-reset)
+
+### GPIOs
+
+- [ ] Provide a GPIO map in a spreadsheet.
+- [ ] Are all MCU peripherals used only once?
+- [ ] If the MCU is held in reset, are the peripheral devices held in their proper state? (Pull-up/pull-downs where needed)
+- [ ] Does the reset line have a proper pull-up or pull-down? (This is a common failure, so worth noting).
+- [ ] Are the debug GPIOs used _only_ for debug purposes?
+
+### Busses
+
+- [ ] Do you have a mechanism to reset all IC's attached to buses (when appropriate)?
+- [ ] Are all I2C devices on the same bus on different addresses?
+- [ ] If your MCU resets in the middle of an I2C read, how do you recover talking with the device you were reading from?[3](#story-3-i2c-resets)
+- [ ] If your MCU resets in the middle of a SPI transfer, how does the device the MCU was talking with handle that?
+- [ ] For all UART devices, are RTS/CTS lines needed?
+- [ ] For all UART devices, are RTS/CTS lines setup properly (this is confusing, because RTS -> CTS, but some devices label them differently)?
+
+### Debug/Test Infrastructure
+
+- [ ] How will you (a firmware engineer) attach a debugger to the board? (By soldering is not usually a good answer).
+- [ ] Are non-essential, but helpful, GPIOs exposed for a debugger, such as SWO or ETM?
+- [ ] Do test points exist for power rails, analog signals, and busses? If not, you'll need a plan for how to debug issues in those areas.
+- [ ] Where will crash logs be stored? (Typically need to survive a system reset). Think about this when thinking about RAM requirements.
```

Comment:
I'm wondering if you might consider another section dedicated to memory with the bottom line being that you should have plenty of space for extra code you'll add in the future, crash data (so coredumps), and logs. Perhaps you have a story about not having enough flash when you wanted it, and you wish you fought harder for a bigger flash part

Reply:
Added this section!